pkg/apis/servicecatalog/v1beta1/types.go

```diff
@@ -833,6 +833,10 @@ type ServiceBindingSpec struct {
 	// namespace that will hold the credentials associated with the ServiceBinding.
 	SecretName string `json:"secretName,omitempty"`
 
+	// List of transformations that should be applied to the credentials returned
```

Comment:
`credentials` is just one field of many that are allowed to come from the broker in a binding. I am wondering if we could remove the assumption that the secret is mapped to the `credentials` field and ask that the operator use a full path for the transform. 

at the moment this allows `[{from:foo, to:bar}]`, given `result.credentials.foo` -> 'secret:{bar:<>}`
but perhaps the more forward looking change could be:`[{from:credentials.foo, to:bar}]`, given `result.credentials.foo` -> 'secret:{bar:<>}`

And if you have this then you can also stick the results of log drain into a secret if the operator wanted to...

Reply:
Yeah, I see your point. Your suggestion is also somewhat related to what I've just noticed in #1869. Since the response schema describes the whole response, the UI (who I expect to be the primary consumer of the response schema) now needs to be aware that the credentials are just one part of the response (and the only one that currently gets into the Secret). My gut feeling tells me this may be problematic. Treating the whole response as being important (instead of just the credentials), should solve this.

Reply:
Given we just tread credentials as secret today, this change is in-line with the current usage. I am just thinking about if we accept this then want to make a follow-up "breaking" transform changes in the future when #1869 is in.

Reply:
I see your point @n3wscott but I don't want to begin unwinding the assumption about 'credentials' here. We don't really have notions for what the OSB (really, the CF) route and log drain payloads mean in kubernetes, and I worry about opening a rathole.

I'm not against figuring out to do with the other types of binding payloads, but I don't want to block this change on it.

I suggest that we:

- Make an issue to figure out what to do with the other types of bindings
- proceed without handling other types of bindings for now
- make an API that is easy for the user to understand (ie, doesn't details with OSB requests and responses they don't know anything about) based on the current API

As we figure out the exact-right behaviors in the future, we can put them in new fields and mark the current fields deprecated.